Upgrade to 0.8.5 results in invalid memory/nil pointer error

**Describe the bug**
After the upgrade from `0.8.4` -> `0.8.5` I now get an error when running kube-vip. 

This is running on a Raspberry Pi 5 (8GB) in a k3s cluster.

```
time="2024-10-31T23:40:49Z" level=info msg="Starting kube-vip.io [v0.8.5]"
time="2024-10-31T23:40:49Z" level=info msg="namespace [kube-system], Mode: [ARP], Features(s): Control Plane:[true], Services:[false]"
time="2024-10-31T23:40:49Z" level=info msg="No interface is specified for VIP in config, auto-detecting default Interface"
time="2024-10-31T23:40:49Z" level=info msg="prometheus HTTP server started"
time="2024-10-31T23:40:49Z" level=info msg="kube-vip will bind to interface [eth0]"
time="2024-10-31T23:40:49Z" level=warning msg="Node name is missing from the config, fall back to hostname"
time="2024-10-31T23:40:49Z" level=info msg="Using node name [pik8s01]"
time="2024-10-31T23:40:49Z" level=info msg="Starting Kube-vip Manager with the ARP engine"
time="2024-10-31T23:40:49Z" level=info msg="Starting UPNP Port Refresher"
time="2024-10-31T23:40:49Z" level=info msg="Beginning cluster membership, namespace [kube-system], lock name [plndr-cp-lock], id [pik8s01]"
I1031 23:40:49.180672       1 leaderelection.go:250] attempting to acquire leader lease kube-system/plndr-cp-lock...
time="2024-10-31T23:40:49Z" level=info msg="Node [pik8s01] is assuming leadership of the cluster"
I1031 23:40:49.193067       1 leaderelection.go:260] successfully acquired lease kube-system/plndr-cp-lock
time="2024-10-31T23:40:49Z" level=info msg="Starting IPVS LoadBalancer"
time="2024-10-31T23:40:49Z" level=info msg="IPVS Loadbalancer enabled for 1.2.1"
time="2024-10-31T23:40:49Z" level=info msg="Kube-Vip is watching nodes for control-plane labels"
time="2024-10-31T23:40:49Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [192.168.6.40/eth0]"
E1031 23:40:49.194773       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 26 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x17eafa0, 0x2e59990})
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/runtime/runtime.go:75 +0x7c
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x400061a1c0?})
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/runtime/runtime.go:49 +0x78
panic({0x17eafa0?, 0x2e59990?})
	/usr/local/go/src/runtime/panic.go:785 +0x124
k8s.io/client-go/kubernetes.(*Clientset).CoreV1(...)
	/go/pkg/mod/k8s.io/client-go@v0.29.1/kubernetes/clientset.go:309
github.com/kube-vip/kube-vip/pkg/cluster.(*Manager).NodeWatcher.func1({{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x1, ...})
	/src/pkg/cluster/clusterLeaderElection.go:314 +0x2c
k8s.io/client-go/tools/cache.(*ListWatch).Watch(0x10?, {{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, 0x0}, 0x0, ...})
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/cache/listwatch.go:111 +0x48
k8s.io/client-go/tools/watch.(*RetryWatcher).doReceive(0x400052a200)
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:102 +0x94
k8s.io/client-go/tools/watch.(*RetryWatcher).receive.func2({0x1e48530, 0x4000414cd0})
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:264 +0x3c
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:259 +0x2c
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x400071a570?)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:226 +0x40
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x40006dbe88, {0x1e2bdc0, 0x400071a570}, 0x0, 0x400040a620)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:227 +0x90
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x40006dbe88, 0x3b9aca00, 0x0, 0x0, 0x400040a620)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:204 +0x80
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext({0x1e48530, 0x4000414cd0}, 0x40006dbf28, 0x3b9aca00, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:259 +0x78
k8s.io/apimachinery/pkg/util/wait.NonSlidingUntilWithContext(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:190
k8s.io/client-go/tools/watch.(*RetryWatcher).receive(0x400052a200)
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:263 +0x220
created by k8s.io/client-go/tools/watch.newRetryWatcher in goroutine 60
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:84 +0x1a4
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0x155bafc]

goroutine 26 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x400061a1c0?})
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/runtime/runtime.go:56 +0xe0
panic({0x17eafa0?, 0x2e59990?})
	/usr/local/go/src/runtime/panic.go:785 +0x124
k8s.io/client-go/kubernetes.(*Clientset).CoreV1(...)
	/go/pkg/mod/k8s.io/client-go@v0.29.1/kubernetes/clientset.go:309
github.com/kube-vip/kube-vip/pkg/cluster.(*Manager).NodeWatcher.func1({{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x1, ...})
	/src/pkg/cluster/clusterLeaderElection.go:314 +0x2c
k8s.io/client-go/tools/cache.(*ListWatch).Watch(0x10?, {{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, 0x0}, 0x0, ...})
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/cache/listwatch.go:111 +0x48
k8s.io/client-go/tools/watch.(*RetryWatcher).doReceive(0x400052a200)
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:102 +0x94
k8s.io/client-go/tools/watch.(*RetryWatcher).receive.func2({0x1e48530, 0x4000414cd0})
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:264 +0x3c
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:259 +0x2c
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x400071a570?)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:226 +0x40
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x40006dbe88, {0x1e2bdc0, 0x400071a570}, 0x0, 0x400040a620)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:227 +0x90
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x40006dbe88, 0x3b9aca00, 0x0, 0x0, 0x400040a620)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:204 +0x80
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext({0x1e48530, 0x4000414cd0}, 0x40006dbf28, 0x3b9aca00, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:259 +0x78
k8s.io/apimachinery/pkg/util/wait.NonSlidingUntilWithContext(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/util/wait/backoff.go:190
k8s.io/client-go/tools/watch.(*RetryWatcher).receive(0x400052a200)
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:263 +0x220
created by k8s.io/client-go/tools/watch.newRetryWatcher in goroutine 60
	/go/pkg/mod/k8s.io/client-go@v0.29.1/tools/watch/retrywatcher.go:84 +0x1a4
```

**To Reproduce**
Steps to reproduce the behavior:
1: Upgrade from `0.8.4` to `0.8.5`
2: Observe Pod status
3: View logs

**Expected behavior**
I would have expected the upgrade to start normally after the version upgrade.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - OS/Distro: Debian 12 Bookworm
 - Kubernetes Version: 1.31.1-k3s
 - Kube-vip Version: 0.8.5

**Additional context**
Here are the environment variable I have set in the `DaemonSet`

```
        env:
        - name: address
          value: 192.168.6.40
        - name: vip_arp
          value: "true"
        - name: lb_enable
          value: "true"
        - name: port
          value: "6443"
        - name: vip_cidr
          value: "32"
        - name: cp_enable
          value: "true"
        - name: cp_namespace
          value: kube-system
        - name: vip_ddns
          value: "false"
        - name: svc_enable
          value: "false"
        - name: vip_leaderelection
          value: "true"
        - name: vip_leaseduration
          value: "15"
        - name: vip_renewdeadline
          value: "10"
        - name: vip_retryperiod
          value: "2"
        - name: prometheus_server
          value: :2112
```

--- a/pkg/cluster/clusterLeaderElection.go
+++ b/pkg/cluster/clusterLeaderElection.go
@@ -75,9 +75,13 @@ func NewManager(path string, inCluster bool, port int) (*Manager, error) {
 		return nil, fmt.Errorf("error creating a new k8s clientset: %v", err)
 	}
 
-	rwConfig := *config
+	rwConfig, err := k8s.NewRestConfig(path, inCluster, hostname)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create k8s REST config for retryClientSet: %w", err)
+	}
+
 	rwConfig.Timeout = 0 // empty value to disable the timeout
-	rwClientSet, err := k8s.NewClientset(&rwConfig)
+	rwClientSet, err := k8s.NewClientset(rwConfig)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create k8s client for retry watcher: %w", err)
 	}

